src/escape/escape.c

```diff
@@ -204,10 +226,32 @@ int main(int argc, char *argv[]) {
                         break;
 
                 case ACTION_UNESCAPE:
+                        e = *i;
```

Comment:
`*i` comes from `argv`, so it should not be freed. It shouldn't be assigned to `e`, which is `_cleanup_free_`. Please define and use a separate `const char*` variable for this value:
```
case ACTION_UNESCAPE: {
      const char *name;

     if (...) {
            r = unit_name_to_instance(*i, &name);
            ...
    } else
           name = *i;
```

Also `char*` variables should be initialized with NULL when they are declared as `_cleanup_free_`. Otherwise, they should not be initialized if they are unconditionally set later.

Reply:
I don’t quite follow – if we’re using `unit_name_to_instance`, `name` should be freed afterwards, right? So I guess it would need to be a `_cleanup_free_` variable after all (though perhaps a separate one, not `e`), and use `strdup` instead of directly assigning `*i`, so that we can `free` in that case as well?